src/analytics/background/periodic.js

```diff
@@ -88,18 +87,17 @@ const attemptPeriodicPing = (
     }
 }
 
+// Schedule all periodic ping attempts at a random minute past the hour, every hour
 const createPeriodicEventJob = period =>
-    new CronJob({
-        cronTime: SCHEDULES.EVERY_HOUR(),
-        start: true,
-        onTick: attemptPeriodicPing(period, getActivePeriodVars(period)),
+    browser.alarms.create(period, {
+        delayInMinutes: Math.floor(Math.random() * 60),
+        periodInMinutes: SCHEDULES.EVERY_HOUR,
     })
 
-// Schedule all periodic ping attempts at a random minute past the hour, every hour
-const jobs = [
-    createPeriodicEventJob('month'),
-    createPeriodicEventJob('week'),
-    createPeriodicEventJob('day'),
-]
+browser.alarms.onAlarm.addListener(alarm =>
+    attemptPeriodicPing(alarm.name, getActivePeriodVars(alarm.name)),
```

Comment:
Just tried it out and it mostly works but you'll need to update this function's signature. Currently it is a higher-order fn of type `(period: string, activePeriodVars: any) => () => Promise<void>` but now as you're using it inside the `addListener` callback it will need to change to `(period: string, activePeriodVars: any) => Promise<void>`, or else all that inner logic won't be invoked (only need to change lines 46-49)